tests/test_curves.py

```diff
@@ -843,3 +843,57 @@ def test_compute_ndarray_error(self):
         c = SplineXYZCurve(X=R * np.cos(phi), Y=R * np.sin(phi), Z=np.zeros_like(phi))
         with pytest.raises(TypeError):
             c.compute("length", grid=np.linspace(0, 1, 10))
+
+    @pytest.mark.unit
+    def test_discontinuous_splines(self):
+        """Test splines that have break points."""
+
+        def test(method, data_key, include_breaks=True):
+            R = 2
+            phi = 2 * np.pi * np.linspace(0, 1, 1001, endpoint=True) ** 2
+
+            discontinuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=break_indices,
+                method=method,
+            )
+
+            continuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=None,
+                method=method,
+            )
+
+            assert discontinuous.method == method
+            assert continuous.method == method
+
+            discon_data = discontinuous.compute(data_key)[data_key]
+            cont_data = continuous.compute(data_key)[data_key]
+
+            if include_breaks:
+                np.testing.assert_allclose(discon_data, cont_data, rtol=1e-3)
+            else:
+                np.testing.assert_allclose(
+                    discon_data[~np.array(break_indices)],
+                    cont_data[~np.array(break_indices)],
+                    rtol=1e-3,
+                )
+            return discon_data, cont_data
+
+        break_indices = [0, 250, 500, 750]
```

Comment:
minor thing but can you move this to above the `def test` so that its clear what it is when its referenced inside that function?

Reply:
Yes I was confused where this variable was being set